testcases/alternativesTests/binaries_test_methods.py

```diff
@@ -180,7 +180,12 @@ def _perform_all_checks(self):
                 if subpackage in self.installed_binaries or subpackage in self.installed_slaves:
                     slaves = self.installed_slaves[subpackage]
                     binaries = self.installed_binaries[subpackage]
-                    tu.passed_or_failed(self, sorted(binaries) == sorted(slaves),
+                    if(sorted(binaries) != sorted(slaves) and subpackage == "devel" and self.rpms.getMajorVersionSimplified() == 8):
+                        diff = tu.two_lists_diff(binaries, slaves)
+                        diff = tu.two_lists_diff(diff, self.installed_slaves["headless"])
+                    else:
+                        diff = []
+                    tu.passed_or_failed(self, diff == [],
```

Comment:
the ojdk8 packages are different, as the headless and devel packages are results of different builds.. due to this fact, the devel package has all the binaries that headless package does but does not list the same slaves for those binaries. This fix allows jsf to check that the binaries and slaves are correct even for ojdk8 (this behaviour is and will always remain a speciality of ojdk8, as in future version this has been already fixed)